### PR TITLE
refactor: Transition from JS to TS

### PR DESCRIPTION
**Pull request by soedirgo**
Mostly satisfying the type checker and changing import/exports. No change to logic or code structure. Also a small fix to the tests.

### CODE REVIEW

File: dist/server.js [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
-var express = require('express');
-var app = express();
+var express = require("express");
 var api = require('./api');
 var project = require('../package.json');
+var app = express();
 app.use(express.json());
 app.use(api);
-app.get('/', function (req, res) {
+app.get('/', function (_req, res) {
```
Comment (kiwicopple) [inline dist/server.js:9]: good Elixir convention :)

Comment (soedirgo) [reply dist/server.js:9]: Oh, it is? I think it was the TS language server yelling at me 😅

File: src/api/extensions.ts [hunk 2]
```diff
@@ -13,4 +15,4 @@ router.get('/', async (req, res) => {
   }
 })
 
-module.exports = router
+export = router
```
Comment (kiwicopple) [inline src/api/extensions.ts:18]: I've never seen this convention before. Should it be `export default router`? or is this a TS thing?

Comment (kiwicopple) [reply src/api/extensions.ts:18]: ah - I think it's not a default export. Usually I'd do something like:

`export const router = Router()`

Comment (soedirgo) [reply src/api/extensions.ts:18]: It's a TS thing yeah. Apparently it's the equivalent of CommonJS's `module.exports` object:
https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

File: dist/server.js (hunk 1 above)
Comment (kiwicopple) [reply dist/server.js:9]: it could be that too, i'm not very familiar with TS

**Comment by kiwicopple** [comment]
OK looks great. I can't remember why I left the `dist` folder in but it makes PR's a bit annoying. I probably thought it would be easier for people to test (just run `node dist server.js`), but I think we should remove it now. Let's merge this then I'll strip it out and add to the gitignore

**Comment by soedirgo** [comment]
👍 Good idea, was about to ask.

**Pull request merged**